### ISSUE TITLE
Help needed with KodiConnector

### ISSUE THREAD

**Issue by nklerk** [nklerk/neeo_driver-kodi#9]
I'm having troubles understanding how i can build a new element for each connection.

I'm calling:
`const ws = new kodiConnector.rpcws(mac, ip);`

It's referring to the following code, Please note its garbage right now as i have tried every think i could think of left and right. I would appreciate if some one want's to help me with this specific file.

https://github.com/nklerk/neeo_driver-kodi/blob/next/devices/kodi/KodiConnector/index.js

@MichaelKohler would you be able to assist me here?

**Comment by MichaelKohler** [comment]
Could you elaborate a bit more on what you are trying to achieve?

**Comment by nklerk** [comment]
I build a constructor for kodi. For every kodi I want to expose the builder, events, send function and in the future maybe more. I can call one function with ‘’’ x=new bla bla’’’ but that doesn’t make all the code in that file unique. I’m having issues with this as I don’t even know the correct wording for it for what I want I’m afraid.

Currently the variables are shared and I don’t want that for this case

**Comment by nklerk** [comment]
@michaelkohler 

I want to define a new kodi connector for every kodi setup (as there can be multiple) so I thought the = new definition would take care of that. Unfortunately if I set a value then it’s set for all, apperantly it’s shared. So I’m looking for a way to define a new kodi connection like a class in other languages. I tried working with prototype but that didn’t work the way I expected. The separate functions I expose must share some variables between the functions but not between the separate kodi instances.

Hope this explanation helps.
